### ISSUE TITLE
 Failed at the angular-phonecat@0.0.0 protractor script.

### ISSUE THREAD

**Issue by ekleiman22** [angular/angular-phonecat#414]
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'protractor' ]
2 info using npm@5.0.3
3 info using node@v8.1.3
4 verbose run-script [ 'preprotractor', 'protractor', 'postprotractor' ]
5 info lifecycle angular-phonecat@0.0.0~preprotractor: angular-phonecat@0.0.0
6 verbose lifecycle angular-phonecat@0.0.0~preprotractor: unsafe-perm in lifecycle true
7 verbose lifecycle angular-phonecat@0.0.0~preprotractor: PATH: C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\GIT\angular-phonecat\node_modules\.bin;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files (x86)\Microsoft ASP.NET\ASP.NET Web Pages\v1.0\;C:\Program Files (x86)\Windows Kits\8.0\Windows Performance Toolkit\;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Users\eug10\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\;c:\Program Files\Microsoft SQL Server\110\DTS\Binn\;c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\ManagementStudio\;c:\Program Files (x86)\Microsoft Visual Studio 10.0\Common7\IDE\PrivateAssemblies\;c:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\MySQL\MySQL Fabric 1.5 & MySQL Utilities 1.5\;C:\Program Files (x86)\MySQL\MySQL Fabric 1.5 & MySQL Utilities 1.5\Doctrine extensions for PHP\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\ManagementStudio\;C:\Program Files\nodejs\;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files\Git\cmd;C:\Users\eug10\AppData\Local\.meteor\;C:\Users\eug10\AppData\Local\Microsoft\WindowsApps;C:\Users\eug10\AppData\Roaming\npm;;C:\Program Files (x86)\Microsoft VS Code\bin
8 verbose lifecycle angular-phonecat@0.0.0~preprotractor: CWD: C:\GIT\angular-phonecat
9 silly lifecycle angular-phonecat@0.0.0~preprotractor: Args: [ '/d /s /c', 'npm run update-webdriver' ]
10 silly lifecycle angular-phonecat@0.0.0~preprotractor: Returned: code: 0  signal: null
11 info lifecycle angular-phonecat@0.0.0~protractor: angular-phonecat@0.0.0
12 verbose lifecycle angular-phonecat@0.0.0~protractor: unsafe-perm in lifecycle true
13 verbose lifecycle angular-phonecat@0.0.0~protractor: PATH: C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\GIT\angular-phonecat\node_modules\.bin;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files (x86)\Microsoft ASP.NET\ASP.NET Web Pages\v1.0\;C:\Program Files (x86)\Windows Kits\8.0\Windows Performance Toolkit\;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Users\eug10\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\;c:\Program Files\Microsoft SQL Server\110\DTS\Binn\;c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\ManagementStudio\;c:\Program Files (x86)\Microsoft Visual Studio 10.0\Common7\IDE\PrivateAssemblies\;c:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\MySQL\MySQL Fabric 1.5 & MySQL Utilities 1.5\;C:\Program Files (x86)\MySQL\MySQL Fabric 1.5 & MySQL Utilities 1.5\Doctrine extensions for PHP\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\ManagementStudio\;C:\Program Files\nodejs\;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files\Git\cmd;C:\Users\eug10\AppData\Local\.meteor\;C:\Users\eug10\AppData\Local\Microsoft\WindowsApps;C:\Users\eug10\AppData\Roaming\npm;;C:\Program Files (x86)\Microsoft VS Code\bin
14 verbose lifecycle angular-phonecat@0.0.0~protractor: CWD: C:\GIT\angular-phonecat
15 silly lifecycle angular-phonecat@0.0.0~protractor: Args: [ '/d /s /c', 'protractor e2e-tests/protractor.conf.js' ]
16 silly lifecycle angular-phonecat@0.0.0~protractor: Returned: code: 199  signal: null
17 info lifecycle angular-phonecat@0.0.0~protractor: Failed to exec protractor script
18 verbose stack Error: angular-phonecat@0.0.0 protractor: `protractor e2e-tests/protractor.conf.js`
18 verbose stack Exit status 199
18 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:283:16)
18 verbose stack     at emitTwo (events.js:125:13)
18 verbose stack     at EventEmitter.emit (events.js:213:7)
18 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:40:14)
18 verbose stack     at emitTwo (events.js:125:13)
18 verbose stack     at ChildProcess.emit (events.js:213:7)
18 verbose stack     at maybeClose (internal/child_process.js:897:16)
18 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)
19 verbose pkgid angular-phonecat@0.0.0
20 verbose cwd C:\GIT\angular-phonecat
21 verbose Windows_NT 10.0.15063
22 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "protractor"
23 verbose node v8.1.3
24 verbose npm  v5.0.3
25 error code ELIFECYCLE
26 error errno 199
27 error angular-phonecat@0.0.0 protractor: `protractor e2e-tests/protractor.conf.js`
27 error Exit status 199
28 error Failed at the angular-phonecat@0.0.0 protractor script.
28 error This is probably not a problem with npm. There is likely additional logging output above.
29 verbose exit [ 199, true ]

**Comment by gkalpak** [comment]
This error message is not informative at all unfortunately. I tried with the same node/npm version (on Windows) and it works fine.

What version of protractor are you using? (E.g. `npm ls protractor`)
Do you get any other error messages (e.g. from protractor)

Things to try:

- Make sure you are serving the app by running `npm run start` on a different terminal/
- Remove `node_modules/` and run `npm install` again.

**Comment by ekleiman22** [comment]
>What version of protractor are you using?
  It was protractor@4.0.14
>Remove node_modules/ and run npm install again.
Done
Then I checked once more version of protractor and got 
 error: Invalid version: "https://registry.npmjs.org/protractor/-/protractor-4.0.14.tgz
So I run
npm install -g protractor
Then 
npm ls protractor 
and got
 Version 5.1.2
Then I run 
npm run protractor
and in the end of all operations got
 Failed at the angular-phonecat@0.0.0 protractor script.
 ERR! This is probably not a problem with npm. There is likely additional logging output above.
Then I added directConnect: true in protractor.conf.js
This time protractor's running was not failed but I got message in Chrome that site not reached.
So I repeated npm start and then npm run protractor. I saw that Chrome was opened then closed
and in command window I got

1 spec, 0 failures
Finished in 2.058 seconds
[18:22:45] I/launcher - 0 instance(s) of WebDriver still running
[18:22:45] I/launcher - chrome #01 passed

So I think that it was success. Thank you very much
